src/hatti/views/dataview.cljs

```diff
@@ -88,6 +88,39 @@
                              {:selected view})
       (put! shared/event-chan {:re-render view}))))
 
+(defn activate-settings-view! [view settings-section]
+  (let [view (keyword view)
+        settings-section (keyword settings-section)
+        views (-> @shared/app-state :views :all)
+        settings-views (-> @shared/app-state :views :settings :all)]
+    (when (contains? (set settings-views) settings-section)
+      (merge-into-app-state! shared/app-state
+                             [:views]
+                             {:selected view})
+      (merge-into-app-state! shared/app-state
+                             [:views :settings]
+                             {:active-tab settings-section})
+      (put! shared/event-chan {:re-render view}))))
+
+(defn activate-integrated-apps-view! [view settings-section app-type]
+  (let [view (keyword view)
+        settings-section (keyword settings-section)
+        settings-views (-> @shared/app-state :views :settings :all)]
+    (when (contains? (set settings-views) settings-section)
+      (transact!
+       shared/app-state
+       (fn [app-state]
+         (-> app-state
+             (assoc-in [:views] {:selected view})
```

Comment:
doesn't `assoc-in` overwrite the values for the key you assoc with? i.e. after this line whatever views had been before, it will now be `{:selected view}`

If so, can we write this with one `assoc` as:

```clojure
(assoc app-state :views {:selected view
                 :settings {:active-tab settings-section
                            :integrated-apps {:active-section app-type}}})
```